### DISCUSSION TITLE
Web application

### DISCUSSION THREAD

**Discussion by rp9823** [iiasa/message_ix#857] [Q&A]
Is there any web application for MESSAGEix?

**Comment by khaeru** [comment]
No, there currently is not.

**Reply by glatterf42** [reply]
But please feel free to share your envisioned use case as an answer here or in a new discussion! Then other people can upvote it and if we recognize that there is a large desire in the community for a web application, we might start developing one. However, please be aware that this is, realistically, on the scale of a few years since there are other issues we consider to be more pressing.

**Reply by rp9823** [reply]
Thanks for your answer. In the paper available at the following link: https://www.sciencedirect.com/science/article/abs/pii/S1364815218302330, they mention an ixmp web application, still in beta version at the time, which has also been used for the IAMC 1.5°C Scenario Explorer hosted by IIASA. I wanted to understand whether it's open source and available or not. For what it may concern my use case, I need to get GDP projections for RCP 4.5, 3.7, 2.6 and 1.9. Even just one of them would be fine. I'm not a "technically versed" user so I was in search for a web app, an interface or a facilitated way to set such scenarios. Let me know if you have any suggestion and thanks again for your help.

**Reply by khaeru** [reply]
What was called the “ixmp web application” in that earlier paper is now called the IIASA **“Scenario Explorer”**. See [Scenario ensembles and database resources](https://iiasa.ac.at/scenario-ensembles-and-database-resources) on the IIASA website for some general information, including a partial list of different Scenario Explorer instances.

- These provide the features you mention, for instance, access to **existing data** for particular measures (such as GDP) for specific scenarios in the context of different projects. Some of these were generated with models like MESSAGEix-GLOBIOM, but others were generated by different models and methods.
- The Scenario Explorer code itself is not open source, but many of the Scenario Explorer instances are open access, meaning you can freely access the data they contain, use them to select, visualize, and/or export the data. In some instances registration is required.

At the moment, this Scenario Explorer does not use the same version of [`ixmp`](https://github.com/iiasa/ixmp/) as the MESSAGEix model (which is developed in the https://github.com/iiasa/message_ix repository, where we are having this discussion). It is entirely separate. To be clear, MESSAGEix is a package that allows to **build new models** and **generate new projections** and, again, there is no web interface for these features.

If all you want to do is access existing data (generated by others using MESSAGE and other models), the Scenario Explorer should suit your needs.
